CMake/resolve_dependency_modules/cpr.cmake

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+include_guard(GLOBAL)
+
+set(VELOX_CPR_VERSION 1.10.5)
+set(VELOX_CPR_BUILD_SHA256_CHECKSUM
+    c8590568996cea918d7cf7ec6845d954b9b95ab2c4980b365f582a665dea08d8)
+set(VELOX_CPR_SOURCE_URL
+    "https://github.com/libcpr/cpr/archive/refs/tags/${VELOX_CPR_VERSION}.tar.gz"
+)
+
+resolve_dependency_url(CPR)
+
+message(STATUS "Building cpr from source")
+FetchContent_Declare(
+  cpr
+  URL ${VELOX_CPR_SOURCE_URL}
+  URL_HASH ${VELOX_CPR_BUILD_SHA256_CHECKSUM})
+set(BUILD_SHARED_LIBS OFF)
+FetchContent_MakeAvailable(cpr)
```

Comment:
The default options seems sensible (except shared libs as discussed above) so this should be fine for now. We might want/need to force the use of existing curl when this is used in combination with some of the adapter deps as they also depend on curl but we can handle that if issues pop up.

Reply:
>We might want/need to force the use of existing curl when this is used in combination with some of the adapter 

@assignUser Got it, thanks. BTW I see a new failure in CI, is it related to this potential issue?

https://app.circleci.com/pipelines/github/facebookincubator/velox/37362/workflows/c8c0f36c-783b-498c-9c6e-f83f7d8ae73e/jobs/249983

```C++
_deps/cpr-src/include/cpr/util.h: In function 'int cpr::util::progressUserFunction(const T*, cpr::cpr_pf_arg_t, cpr::cpr_pf_arg_t, cpr::cpr_pf_arg_t, cpr::cpr_pf_arg_t)':
_deps/cpr-src/include/cpr/util.h:26:36: error: 'CURL_PROGRESSFUNC_CONTINUE' was not declared in this scope
   26 |     static_assert(cancel_retval != CURL_PROGRESSFUNC_CONTINUE);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: error: unrecognized command line option '-Wno-stringop-overread' [-Werror]
```

Reply:
Well looking at the linux-adapters job this might have already proven an issue ^^) https://app.circleci.com/pipelines/github/facebookincubator/velox/37362/workflows/c8c0f36c-783b-498c-9c6e-f83f7d8ae73e/jobs/249983?invite=true#step-108-42957_71

My first guess would be that the system openssl is linked against a different curl version than the one bundled with cpr. So setting `CPR_USE_SYSTEM_CURL=TRUE` when building adapters might be the solution to that.

Reply:
Got it, `CPR_USE_SYSTEM_CURL` is OFF by default in cpr, I will set it as ON in `cpr.camke`, thanks a lot.

Reply:
@assignUser The same error pop up, even after adding the `set(CPR_USE_SYSTEM_CURL ON)`, could you please help to take a look?